### ISSUE TITLE
Home Assistant cast shows generated view instead of selected

### ISSUE THREAD

**Issue by pszafer** [home-assistant/frontend#3778]
**Home Assistant release with the issue:**
0.99.2

**Last working Home Assistant release (if known):**

**UI (States or Lovelace UI?):**
HA Cast
**Browser and Operating System:**
not applicable

**Description of problem:**
My full story:
- I upgraded my HA to 0.99.2 and tested cast to nvidia shield and callled service:
```
cast.show_lovelace_view
{
  "entity_id": "media_player.shield",
  "view_path": "camera"
}
```
It was working.

- Upgraded HA in my parents house and I wanted to test on same shield (we share vpn network with same subnet).
It was working
- Now calling same service from my instance of HA is not working
I get:
```
Unable to find a view with path camera
```

if I call service with `view_path: 0` I can see display with States UI of HA.
I can't find way to revert to see lovelace in cast.

My configuration yaml of http:
```
http:
  server_port: 443
  ssl_certificate: acme.fullchain
  ssl_key: acme.key
  base_url: https://myurl.com
  use_x_forwarded_for: true
  trusted_proxies:
    - HAPROXY IP
```

PS
Is it possible to control UI via remote controller of android tv?

**Comment by bramkragten** [comment]
Cast doesn't support states, it only has Lovelace. Are you sure the path exists? Can you visit `/lovelace/camera`

When you use `view_path: 0` your first lovelace view is shown.

**Comment by pszafer** [comment]
I know it suppose to support lovelace only, but I think I see states...
Lovelace path `/lovelace/camera` exists.

I provide screenshots for you and photo:
![image](https://user-images.githubusercontent.com/690510/65459073-4e5cd480-de4f-11e9-91e3-47d142a132f2.png)
(you can see url in the top)

![image](https://user-images.githubusercontent.com/690510/65459196-a7c50380-de4f-11e9-9799-b5df8c259ca8.png)
(this is states)

and this is photo from phone
![20190923_221143](https://user-images.githubusercontent.com/690510/65459311-e8248180-de4f-11e9-9a08-3f50e2d7b77f.jpg)

it's different order because Nvidia shield is providing different language than browser in my pc but I think it is states. I will try to install vysor and made photo of scrolled down screen somehow.

**Comment by bramkragten** [comment]
Could be autogenerated lovelace, cast will generate that it can't get your lovelace config.

**Comment by pszafer** [comment]
Ok, might be autogenerated lovelace.
Just another photo. View is identical as states.

![20190923_224203](https://user-images.githubusercontent.com/690510/65461236-d513b080-de53-11e9-9075-0cbd59b06154.jpg)

So, how can I investigate why cast can't get my lovelace config?

**Comment by mf-social** [comment]
Exactly the same thing happening here - got a nest hub today and started setting it up - I'm on 103.5

First try to set it up and sending `view_path: '0'` (if I don't put the quotes around the zero it complains it isn't a string) worked fine and was sending my lovelace first page.  Sent '1' and '2' etc, no problems.  sent '0' again, still no problems.

Rebooted homeassistant to do a few bits, when it came back up I sent the exact same request (was already pre-populated on the page) and it now comes up as states_ui (or, as mentioned above, possibly it is 'default lovelace').  Sending anything other than '0' results in a page not found message on the nest hub. (Interesting to note that it says this if I use the page names as well, so like 'home', 'people' as defined in lovelace)

Nothing changed in lovelace between reboots.

Please fix, it was really good for the 5 minutes in was working before the reboot 😢

**Comment by codename-osni** [comment]
Same issue here, except I never rebooted Home Assistant.
I casted one of my views, (battery), tried to cast another view (maps) which didn't work, then tried the battery view again, and it doesn't work anymore.

The only view that I'm able to cast right now is the default_view, which is the automatically generated one that has all the entities.
If I navigate to /lovelace/default_view on my web browsers, it's a totally different (and correct) one :(

PS: Not using Nabu Casa.
Also, https://cast.home-assistant.io/ did not work for me.

**Comment by gtjj** [comment]
Same issue here:

I am using the Nabu Casa Cloud URL.

Can only cast `default_view` (even after renaming the view, which is weird).

On Cast.home-assistant.io, i only see the "Home" view. When i cast that i get the page with a bunch of entities, like the screenshots provided by the users above.

This is one of the coolest HA features if we can get it to work. Please take a look at this problem!

**Comment by Phibins** [comment]
Same issue here:

I am using the Nabu Casa Cloud URL.

Can only cast default_view (even after renaming the view, which is weird).

On Cast.home-assistant.io, i only see the "Home" view. When i cast that i get the page with a bunch of entities, like the screenshots provided by the users above.

This is one of the coolest HA features if we can get it to work. Please take a look at this problem!

**Comment by walthowd** [comment]
Same issue as described for me, HA 106.3 going to Chromecast.

**Comment by joshmd** [comment]
Seeing the same issue, I've never been able to get cast working consistently on my nest hub

**Comment by blowk** [comment]
The problem still is HA 0.110.2. The Lovelace view_path are incorrect when sending it with the Google Cast function. Only the default_view can be showed.
I haven't found a solution how to fix it.

When I use the website https://cast.home-assistant.io/, then I can select the Lovelace views.
Is the website different then Google Cast (https://www.home-assistant.io/integrations/cast/ - https://github.com/home-assistant/core/tree/dev/homeassistant/components/cast) ?

**Comment by michaelblight** [comment]
I have found that it (hopefully) reliably works for me if I send three service calls: turn off; then cast default_view; then cast the view I want.

I had also blocked the hub's access to 4.4.4.4 and 8.8.8.8 as per some other suggestion. However, when I unblocked it, it continued to work.

The main problem now seems to be it only lasts about 10 minutes and then reverts to the clock/photos, so I have to keep re-casting.
